go/tricorder/messages/api.go

```diff
@@ -42,6 +72,58 @@ type Distribution struct {
 	Ranges []*RangeWithCount `json:"ranges,omitempty"`
 }
 
+// Duration represents a duration of time
+// For negative durations, both Seconds and Nanoseconds are negative.
+type Duration struct {
+	Seconds     int64
+	Nanoseconds int32
+}
+
+func NewDuration(d time.Duration) Duration {
+	return newDuration(d)
+}
+
+// SinceEpoch returns the amount of time since unix epoch
+func SinceEpoch(t time.Time) Duration {
+	return sinceEpoch(t)
+}
+
+// AsGoDuration converts this duration to a go duration
+func (d Duration) AsGoDuration() time.Duration {
+	return d.asGoDuration()
+}
+
+// AsGoTime Converts this duration to a go time.
+// This is the inverse of SinceEpoch.
+func (d Duration) AsGoTime() time.Time {
+	return d.asGoTime()
+}
+
+func (d Duration) String() string {
+	return d._string()
+}
+
+// RpcValue represents the value of a metric for go rpc.
+type RpcValue struct {
+	// The value's type
+	Kind types.Type
```

Comment:
You could just store an interface{} and then do type conversion to grab the data.